### ISSUE TITLE
Slow-starting kernel displays name of 'No Kernel!' until startup completes

### ISSUE THREAD

**Issue by kevin-bates** [jupyterlab/jupyterlab#6007]
**Describe the bug**
Slow-starting kernels display "No Kernel!" as the kernel name until the kernel has started.  This can be confusing to the end-user since they believe they've started a valid kernel, and is a deviation from classic notebook, which uses the _status circle_ to indicate the kernel is starting.

**To Reproduce**
I'm afraid this may require truly remote kernels.  If I modify a local kernelspec to delay the kernel's startup, I suspect the fact that the connection file is still valid (due to local ports) enables the server to behave as if the kernel has started (since Popen essentially spawns a background process).  

With Enterprise Gateway, kernels are launched into resource-managed clusters.  In this case, the kernel runs in YARN cluster mode and is launched via spark-submit.  As a result, the delay to convey connection information is right about 10 seconds.  

Here are some applicable log entries.  (Note, I'm running with a [docker image](https://hub.docker.com/r/elyra/nb2kg) built with Lab and NB2KG extensions.  As a result `NotebookApp` appears in the log entries despite this being Lab code.)

```
[I 21:48:57.029 NotebookApp] Request start kernel: kernel_id=None, path="YARN"
[I 21:49:07.255 NotebookApp] Kernel started: 52dbc365-9bb5-41ea-9a17-51412f8e74cb
[I 21:49:07.274 NotebookApp] Kernel retrieved: {'connections': 0, 'last_activity': '2019-02-19T21:49:07.091396Z', 'execution_state': 'starting', 'id': '52dbc365-9bb5-41ea-9a17-51412f8e74cb', 'name': 'spark_python_yarn_cluster'}
```

**Expected behavior**
It would be nice to see the display-name from the kernelspec instead of `No Kernel!`

**Screenshots**
Here's the screen shot taken during startup:
![kernelstarting_nokernel](https://user-images.githubusercontent.com/22599560/53051118-e9bc4000-344f-11e9-9734-3fec1d7bb2ff.png)
 and after:
![kernelstarted_displayname](https://user-images.githubusercontent.com/22599560/53051167-02c4f100-3450-11e9-8ddf-ffa024cd86a3.png)

**Desktop (please complete the following information):**

- OS: MacBook Pro (Darwin 18.2.0), Container OS: GNU/Linux 
- Browser: Chrome Version 72.0.3626.109
- JupyterLab: jupyterlab  0.35.4, jupyterlab-launcher 0.13.1, jupyterlab-server 0.2.0

**Comment by jasongrout** [comment]
Thanks. Let's take a look to see if this is something we can do before 1.0.

**Comment by kevin-bates** [comment]
That would be excellent Jason.  I'm sorry I don't have the expertise to contribute the solution but I'd be happy to help test since I have an environment in which this is easily viewable.

**Comment by ellisonbg** [comment]
Other notes from recent discussions:

* We should also disable shift-enter or give users some indicator that they can't run code until the kernel has started.
* This is also relevant for consoles, and potentially terminals.

**Comment by jasongrout** [comment]
The status indication is solved in #7252 - now kernels that are starting show as "initializing".

**Comment by kevin-bates** [comment]
Thank you @jasongrout!
